Solar system activity

gupta2140 opened the pull request:
The appearance of the main Solar System model for vertical screen which is presented at the start of the activity, is yet to be decided. However, other layouts are set for vertical screens as well. Some animations are yet to be added, like when transisting from solar model to the quiz screen and vice-versa, as there in Categorization activity. The real images of the planets are of very low resolution and hence not clear. They'll be changed.

petitlapin commented on src/activities/solar_system/ActivityInfo.qml:
```diff
@@ -0,0 +1,40 @@
+  author: "Aman Kumar Gupta &lt;gupta2140@gmail.com&gt;"
+  demo: true
+  //: Activity title
+  title: "Solar System"
```
qsTr

petitlapin commented on src/activities/solar_system/ActivityInfo.qml:
```diff
@@ -0,0 +1,40 @@
+  //: Help prerequisite
+  prerequisite: ""
+  //: Help manual
+  manual: qsTr("Click on a planet or the Sun to reveal questions. Each question contains 4 options. One of those is 100% correct. 				Try to answer the questions till you get a 100% closeness in the closeness meter.")
```
why the spaces?

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,354 @@
+		{
+			"clipImg": "qrc:/gcompris/src/activities/solar_system/resource/sun_clip.svg",
+			"realImg": "qrc:/gcompris/src/activities/solar_system/resource/sun_real.svg",
+            "bodyName": qsTr("Sun"),
```
indentation of the file

petitlapin commented on src/activities/solar_system/PlanetInSolarModel.qml:
```diff
@@ -0,0 +1,80 @@
+    height: width
+
+    property string planetImgClipSrc
+    property string nameOFThePlanet
```
naming convention for both

petitlapin commented on src/activities/solar_system/PlanetInSolarModel.qml:
```diff
@@ -0,0 +1,80 @@
+    property string planetImgClipSrc
+    property string nameOFThePlanet
+
+    GCText{
```
missing space

petitlapin commented on src/activities/solar_system/QuizScreen.qml:
```diff
@@ -0,0 +1,230 @@
+        anchors.margins: 10 * ApplicationInfo.ratio
+
+        Item {
+            width: (items.bar.level === 3) ? 0
```
why not set visible if level === 3 instead of setting no width/height?

petitlapin commented on src/activities/solar_system/QuizScreen.qml:
```diff
@@ -0,0 +1,230 @@
+                sourceSize.width: Math.min(parent.width, parent.height) * 0.9
+                anchors.centerIn: parent
+                source: mainQuizScreen.planetRealImage
+                visible: items.bar.level != 3
```
no need if you set the visibility of the item

petitlapin commented on src/activities/solar_system/QuizScreen.qml:
```diff
@@ -0,0 +1,230 @@
+            anchors.centerIn: parent
+            color: "black"
+            fontSize: background.horizontalLayout ? mediumSize : smallSize
+            text: "Closeness: " + closenessValueInMeter
```
qsTr + arg

petitlapin commented on src/activities/solar_system/QuizScreen.qml:
```diff
@@ -0,0 +1,230 @@
+            anchors.centerIn: parent
+            color: "black"
+            fontSize: background.horizontalLayout ? mediumSize : smallSize
+            text: "Hint: " + hint
```
qsTr +arg

petitlapin commented on src/activities/solar_system/Solar_system.qml:
```diff
@@ -0,0 +1,147 @@
+                    Activity.showSolarModel()
+            }
+            onHintClicked: {
+                if(items.mainQuizScreen.hintVisible)
```
items.mainQuizScreen.hintVisible = !items.mainQuizScreen.hintVisible

gupta2140 commented:
Made the changes as per comments. Please review whenever free.

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,349 @@
+                        { 	//sub-level 3
+                            "question": qsTr("Sun is ___ years old"),
+                            "options": [qsTr("1.2 billion"),qsTr("3 billion"),qsTr("7 billion"),qsTr("4.5 billion")],
+                            "closeness": ["1%","55%","25%","100%"]
```
do we really want to display 0.1% differences?

gupta2140 replied:
umm....i'm not quite sure of it. But 0.1% is too small value when represented in the closeness meter, so I think in that point of view, we may ignore it.

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,349 @@
+                        },
+                        { 	//sub-level 4
+                            "question": qsTr("The maximum temperature on Mercury is"),
+                            "options": [qsTr("145 Degrees"),qsTr("35  Degrees"),qsTr("427  Degrees"),qsTr("273  Degrees")],
```
double spaces between number and degrees

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,349 @@
+                        },
+                        { 	//sub-level 2
+                            "question": qsTr("How long does it take Neptune to make one revolution around the Sun?"),
+                            "options": [qsTr("165 Years"),qsTr("3 Years"),qsTr("100 Years"),qsTr("1 Years")],
```
1 year

petitlapin commented on src/activities/solar_system/QuizScreen.qml:
```diff
@@ -0,0 +1,205 @@
+            anchors.centerIn: parent
+            color: "black"
+            fontSize: background.horizontalLayout ? mediumSize : smallSize
+            text: qsTr("Closeness: ") + closenessValueInMeter
```
.arg

petitlapin commented on src/activities/solar_system/Solar_system.qml:
```diff
@@ -0,0 +1,246 @@
+                        text: qsTr("No closeness meter and no hint")
+                        checked: background.assessmentMode
+                        onCheckedChanged: {
+                                background.assessmentMode = checked
```
it should be changed only when you save the data to avoid discrepancy

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,349 @@
+                        { 	//sub-level 1
+                            "question": qsTr("The sun is the_____ largest object in our solar system"),
+                            "options": [qsTr("Sixth"),qsTr("Third"),qsTr("First"),qsTr("Seventh")],
+                            "closeness": ["17.5%","67%","100%","1%"]
```
I think it would be better to only store the integer values instead of the string with the percent. Is it much change?

gupta2140 replied:
No, its not much change, i'll do it in the next commit :)

gupta2140 commented:
Made all the changes as per comments. Waiting for review.

gupta2140 commented:
Planet name text position adjusted for vertical screens, added animations for appearance and disappearance of the quiz screen and added better resolution images of some planets. Waiting for review :)

petitlapin commented:
you need to remove the execute rights for the image saturn_real.png (chmod -x saturn_real.png)

I don't see the interest of the hint. Just displaying the planet order does not help (except for the question about the order).

There is some overriding elements in vertical mode.

Can you randomize the answer positions?

Johnny

gupta2140 commented:
The positions of the options are now set to be shuffled and the overriding elements in vertical screens are now fixed. Waiting for review :)

gupta2140 commented:
Planet floating animations added.

gupta2140 commented:
- Hint for answers of questions based on order of planets can be found out by visiting the solar system in the "Hint" option.
- Hint for questions on Maximum and minimum temperature, duration of year are provided.

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                        { 	//sub-level 1
+                            "question": qsTr("The sun is the_____ largest object in our solar system"),
+                            "options": [qsTr("Sixth"),qsTr("Third"),qsTr("First"),qsTr("Seventh")],
+                            "closeness": [17.5,67,100,1]
```
add a space after the "," to ease the readiness

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                    "bodyName": qsTr("Sun"),
+                    "levels": [
+                        { 	//sub-level 1
+                            "question": qsTr("The sun is the_____ largest object in our solar system"),
```
can you update the question to not have the ____ in the middle?

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                            "closeness": [1,55,25,100]
+                        },
+                        { 	//sub-level 4
+                            "question": qsTr("How long does it take the Sun’s light to reach the earth?"),
```
Earth

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                            "closeness": [17.5,67,100,1]
+                        },
+                        { 	//sub-level 2
+                            "question": qsTr("The temperature of the sun is around _____degrees"),
```
Sun

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                    "realImg": "qrc:/gcompris/src/activities/solar_system/resource/mercury_real.svg",
+                    "bodyName": qsTr("Mercury"),
+                    "temperatureHint": qsTr("The maximum temperature on Earth is 58 degrees."),
+                    "lengthOfYearHint": qsTr("The length of an year on Venus is 225 days."),
```
a year

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                            "closeness": [4.8,1,100,61]
+                        },
+                        { 	//sub-level 5
+                            "question": qsTr("How many days makes an year on Mercury ?"),
```
no space between Mercury and ?

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                        },
+                        { 	//sub-level 5
+                            "question": qsTr("How many days makes an year on Mercury ?"),
+                            "options": [qsTr("365 Days"),qsTr("433 Days"),qsTr("88 Days"),qsTr("107 Days")],
```
days, no uppercase

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                        },
+                        { 	//sub-level 2
+                            "question": qsTr("Venus is as heavy as"),
+                            "options": [qsTr("0.7 Earths"),qsTr("0.8 Earths "),qsTr("1.3 Earths "),qsTr("2.5 Earths ")],
```
no need of spaces at the end

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                        },
+                        { 	//sub-level 4
+                            "question": qsTr("How long is a day on Earth?"),
+                            "options": [qsTr("12 Hours"),qsTr("24 Hours"),qsTr("365 Hours"),qsTr("48 Hours")],
```
no uppercase

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                            "closeness": [100,60.4,1,80]
+                        },
+                        { 	//sub-level 2
+                            "question": qsTr("Saturn is the ______ largest  planet in the Solar System"),
```
double space between largest and planet

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                        },
+                        { 	//sub-level 2
+                            "question": qsTr("How many years does it take Uranus to go once around the Sun?"),
+                            "options": ["1 years","24 years","68 years","84 years"],
```
1 year

petitlapin commented on src/activities/solar_system/QuizScreen.qml:
```diff
@@ -0,0 +1,221 @@
+        anchors.top: parent.top
+    }
+
+    Rectangle {
```
this is behind the bar, is there a way to move it so it won't be overriden by it?

gupta2140 replied:
Fixed. :+1:

petitlapin commented on src/activities/solar_system/Solar_system.qml:
```diff
@@ -0,0 +1,340 @@
+        property int itemWidth: horizontalLayout ? background.width / 9 : (background.height - bar.height) / 9
+
+        //Arrangement of all the planets in the solar system
+        GridView{
```
missing space

petitlapin commented on src/activities/solar_system/Solar_system.qml:
```diff
@@ -0,0 +1,340 @@
+            property var hint1
+            property var hint2
+
+            title: "Hint"
```
qsTr

petitlapin commented on src/activities/solar_system/Solar_system.qml:
```diff
@@ -0,0 +1,340 @@
+
+        //Hint dialog while playing the quiz
+        DialogBackground {
+            id: hintDialog
```
if you resize the screen, the button overrides the text

gupta2140 replied:
fixed :+1:

petitlapin commented on src/activities/solar_system/solar_system.js:
```diff
@@ -0,0 +1,112 @@
+function start(items_) {
+    items = items_
+    currentLevel = 0
+    dataset= Dataset.get()
```
space

petitlapin commented on src/activities/solar_system/solar_system.js:
```diff
@@ -0,0 +1,112 @@
+    items.quizScreenVisible = true
+    currentPlanetLevels = dataset[index].levels
+    items.mainQuizScreen.planetRealImage = dataset[index].realImg
+    items.hintDialog.hint1 = "1. The <b>farther</b> a planet from the Sun, the <b>lower</b> is it's temperature.<br><font color=\"#3bb0de\">%1</font>".arg(dataset[indexOfSelectedPlanet].temperatureHint)
```
qsTr()
Is this text displayed? (we always use the one from the dataset?)

gupta2140 replied:
Yes, this text is displayed along with the ones in the dataset.
The text here, tells us the proportionate variation of the properties among planets (common for all the planets), and the ones in the dataset gives the datas which can be used as reference values to reach the answer(differ with planets).

petitlapin replied:
Then, it misses the qsTr() :)

gupta2140 replied:
yes, I had added it along with all the other changes in the reviews, yesterday in the commits below ;)

gupta2140 commented:
Fixed all overriding issues. Waiting for review!

petitlapin commented on src/activities/solar_system/Dataset.js:
```diff
@@ -0,0 +1,368 @@
+                            "closeness": [100, 60.4, 1, 80]
+                        },
+                        { 	//sub-level 2
+                            "question": qsTr("How large is Staurn compared to other planets in the Solar System?"),
```
Saturn

gupta2140 replied:
Done :+1:

gupta2140 commented:
Sorry for mistakes in commit messages. Light was turned off and some keys in my keyboard don't work properly :(
